x/leverage/keeper/oracle.go

```diff
@@ -51,7 +57,23 @@ func (k Keeper) TokenDefaultDenomPrice(ctx sdk.Context, baseDenom string) (sdk.D
 		return sdk.ZeroDec(), t.Exponent, types.ErrBlacklisted
 	}
 
-	price, err := k.oracleKeeper.GetExchangeRate(ctx, t.SymbolDenom)
+	var price sdk.Dec
+
+	if historic && minHistoracleStamps > 0 {
+		// historic price
+		var numStamps uint32
+		price, numStamps, err = k.oracleKeeper.MedianOfHistoricMedians(ctx, t.SymbolDenom, minHistoracleStamps)
+		if err == nil && numStamps < uint32(minHistoracleStamps) {
+			return sdk.ZeroDec(), t.Exponent, types.ErrNoHistoricMedians.Wrapf(
+				"requested %d, got %d",
+				minHistoracleStamps,
+				numStamps,
+			)
+		}
+	} else {
+		// current price
+		price, err = k.oracleKeeper.GetExchangeRate(ctx, t.SymbolDenom)
```

Comment:
why do we return a current price when `historic=true && minHistoracleStamps = 0`?

Reply:
`minHistoracleStamps` is going to be the parameter which determines how many medians the leverage module asks for - but if we're asking for `0` medians, we don't want to ask the oracle module to compute a median from a slice of zero elements.

Once I add that parameter in properly, I will use earlier if statements to stop `historic == true` calls to price functions when `minHistoracleStamps == 0`